### PR TITLE
feat(core): Enriched Bans On Vo

### PR DESCRIPTION
**Pull request by mattjoke**
* created EnrichedBansOnVo
* added getEnrichedBansForVo function
* added getEnrichedBansForUser function

### CODE REVIEW

File: perun-base/src/main/resources/perun-roles.yml [hunk 1]
```diff
@@ -6801,6 +6801,27 @@ perun_policies:
     include_policies:
       - default_policy
 
+  getEnrichedBansForVo_int_List<String>_policy:
+    policy_roles:
+      - PERUNOBSERVER:
+      - GROUPADMIN: Vo
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:6807]: Not sure about GROUPADMIN and GROUPOBSERVER, check with @bodnara which roles we need.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/VosManagerEntry.java [hunk 2]
```diff
@@ -876,4 +879,30 @@ public void setVosManagerBl(VosManagerBl vosManagerBl)
...
+		Utils.checkPerunSession(sess);
+
+		Vo vo = vosManagerBl.getVoById(sess, voId);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getEnrichedBansForVo_int_List<String>_policy", vo)) {
+			throw new PrivilegeException(sess, "getEnrichedBansForVo");
+		}
+
+		return vosManagerBl.getEnrichedBansForVo(sess, voId, attrNames);
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/VosManagerEntry.java:893]: Filter the attributes before returning.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/VosManagerEntry.java [hunk 3]
```diff
@@ -876,4 +879,30 @@ public void setVosManagerBl(VosManagerBl vosManagerBl)
...
+		Utils.checkPerunSession(sess);
+
+		User user = perunBl.getUsersManagerBl().getUserById(sess, userId);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getEnrichedBansForUser_int_List<String>_policy", user)) {
+			throw new PrivilegeException(sess, "getEnrichedBansForUser");
+		}
+
+		return vosManagerBl.getEnrichedBansForUser(sess, userId, attrNames);
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/VosManagerEntry.java:906]: Same here

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java [hunk 4]
```diff
@@ -760,6 +760,28 @@ public List<BanOnVo> call(ApiCaller ac, Deserializer parms) throws PerunExceptio
 		}
 	},
 
+	getEnrichedBansForVo {
```
Comment (xflord) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java:763]: add Javadocs

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java [hunk 5]
```diff
@@ -760,6 +760,28 @@ public List<BanOnVo> call(ApiCaller ac, Deserializer parms) throws PerunExceptio
...
+		public List<EnrichedBanOnVo> call(ApiCaller ac, Deserializer params) throws PerunException {
+			List<String> attrNames = null;
+			if (params.contains("attrNames")) {
+				attrNames = params.readList("attrNames", String.class);
+			}
+			return ac.getVosManager().getEnrichedBansForVo(ac.getSession(), params.readInt("vo"), attrNames);
+		}
+	},
+
+	getEnrichedBansForUser{
```
Comment (xflord) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java:774]: Here as well

**Review by xflord — CHANGES_REQUESTED** [review]

**Review by Johaney-s — COMMENTED** [review]
good job, i have some cosmetic, mainly java-:dog: suggestions

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java [hunk 6]
```diff
@@ -760,6 +760,45 @@ public List<BanOnVo> call(ApiCaller ac, Deserializer parms) throws PerunExceptio
...
+			}
+			return ac.getVosManager().getEnrichedBansForVo(ac.getSession(), params.readInt("vo"), attrNames);
+		}
+	},
+
+	/**
+	 * Get list of all EnrichedBans for User.
+	 * @param user int user id
+	 * @param attrNames List<String> list of attributes names, if null or empty, returns all attributes
+	 * @return List<BanOnVo> vo bans for given group
```
[the remark this answers is not in the document]
Comment (mattjoke) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java:787]: Shouldn't there be a `vo bans for given user`?

[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java]: of course :+1:

File: perun-utils/rpc-methods-javadoc-generator/parseRpcMethods.pl [hunk 7]
```diff
@@ -294,6 +294,10 @@
 $objectExamples{"List&lt;EnrichedBanOnFacility&gt;"} = $listPrepend . $objectExamples{"EnrichedBanOnFacility"} . $listAppend;
 $objectExamples{"List<EnrichedBanOnFacility>"} = $objectExamples{"List&lt;EnrichedBanOnFacility&gt;"};
 
+$objectExamples{"EnrichedBanOnVo"} = "{ \"vo\" : " . $objectExamples{"Vo"} . " , \"member\" : " . $objectExamples{"RichMember"} . " , \"banOnVo\" : " . $objectExamples{"BanOnVo"} . " }";
```
Comment (Johaney-s) [inline perun-utils/rpc-methods-javadoc-generator/parseRpcMethods.pl:297]:
```suggestion
$objectExamples{"EnrichedBanOnVo"} = "{ \"vo\" : " . $objectExamples{"Vo"} . " , \"member\" : " . $objectExamples{"RichMember"} . " , \"ban\" : " . $objectExamples{"BanOnVo"} . " }";
```

**Review by Johaney-s — APPROVED** [review]

**Review by xflord — APPROVED** [review]

**Pull request merged**